features/eif-adapter/src/main/java/org/opennms/features/eifadapter/EifParser.java

```diff
@@ -0,0 +1,183 @@
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.eifadapter;
+
+import com.google.common.net.InetAddresses;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.events.EventBuilder;
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.netmgt.xml.event.Parm;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EifParser {
+
+    public static final Logger LOG = LoggerFactory.getLogger(EifParser.class);
+    private static final int eifStartOffset = 37;
+    enum m_eifSeverities { FATAL, CRITICAL, MINOR, WARNING, OK, INFO, HARMLESS, UNKNOWN }
+
+    public static List<Event> translateEifToOpenNMS(NodeDao nodeDao, StringBuilder eifBuff) {
+
+        // Create a list of events to return to the packet processor
+        List<Event> translatedEvents = new ArrayList<>();
+
+        // Loop over the received EIF package until we run out of events
+        while(eifBuff.length() > 0 && eifBuff.indexOf(";END") > 1) {
+            // Extract a single event from the package
+            int eventStart = eifBuff.indexOf("<START>>");
+            int eventEnd = eifBuff.indexOf(";END");
+            String eifEvent = eifBuff.substring(eventStart + eifStartOffset,eventEnd);
+            eifBuff.delete(0,eventEnd+4);
+
+            // Parse the EIF slots into OpenNMS parms, and try to look up the source's nodeId
+            String eifClass = eifEvent.split(";")[0];
+            String eifSlots = eifEvent.substring(eifClass.length()+1,eifEvent.length()).
+                    replaceAll(System.getProperty("line.separator"),"");
+            Map<String, String> eifSlotMap = parseEifSlots(eifSlots);
+            List<Parm> parmList = new ArrayList<>();
+            eifSlotMap.entrySet().forEach(p -> parmList.add(new Parm(p.getKey(),p.getValue())));
+            long nodeId = connectEifEventToNode(nodeDao, eifSlotMap);
+
+            // Add the translated event to the list
+            translatedEvents.add(
+                    new EventBuilder("uei.opennms.org/vendor/IBM/EIF/"+eifClass,"eif").
```

Comment:
`opennms-base-assembly/src/main/filtered/etc` contains the pristine version of $OPENNMS_HOME/etc.

In this case, we would add a new event file in the *events* subfolder, and include it in the default *eventconf.xml*.